src/main/java/org/elasticsearch/search/aggregations/bucket/BestDocsDeferringCollector.java

```diff
@@ -189,7 +263,7 @@ public void replayRelatedMatches(ScoreDoc[] sd) throws IOException {
                 if ((rebased >= 0) && (rebased <= maxDocId)) {
                     currentScore = scoreDoc.score;
                     currentDocId = rebased;
-                    leafCollector.collect(rebased, 0);
+                    leafCollector.collect(rebased, scoreDoc.shardIndex);
```

Comment:
Could we add a comment here reminding that the shardIndex is being used to store the parent bucket ordinal?